dataStructures/queues/queue_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestNewOrder(t *testing.T) {
+	cases := []struct{
+		name, product, customerName string
+		priority, quantity int
+		want *Order
+	}{
+		{"create new order", "macbook", "andela", 1, 20, &Order{1, 20, "macbook", "andela"}},
+		{"create another order", "windows", "apple", 2, 30, &Order{2, 30, "windows", "apple"}},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ord := newOrder(tc.priority, tc.quantity, tc.product, tc.customerName)
+			if ord.customerName != tc.want.customerName {
+				t.Errorf("got %s, want %s", ord.customerName, tc.want.customerName)
+			}
+
+			if ord.product != tc.want.product {
+				t.Errorf("got %s, want %s", ord.product, tc.want.product)
+			}
+
+			if ord.priority != tc.want.priority {
+				t.Errorf("got %d, want %d", ord.priority, tc.want.priority)
+			}
+
+			if ord.quantity != tc.want.quantity {
+				t.Errorf("got %d, want %d", ord.quantity, tc.want.quantity)
+			}
+		})
+	}
+}
+
+func TestAddToQueue(t *testing.T) {
```

Comment:
I think you need more comprehensive tests because they're not picking up a bug I think you have. You can ignore the extra information like customer and product for now because you're testing priority ordering.

Try this:
```Go
cases := []struct{
  name string
  priorities []int // Order quantities will be equal to their index in this slice
  wantQuantities []int
}{
  {
    name: "multiple equal priorities",
    priorities: []int{3, 8, 2, 5, 3, 3, 2, 3, 2},
    wantQuantities: []int{2, 6, 8, 0, 4, 5, 7, 3, 1}, // do you see what I've done here? It's an argsort https://stackoverflow.com/questions/17901218/numpy-argsort-what-is-it-doing
  },
}

for _, tc := range cases {
  t.Run(tc.name, func(t *testing.T){
    // Make a new Queue for every test so that they don't overwrite each other.
    // This makes it easier to debug because you only have to think of a single test at a time.
    q := &Queue{}

    var orders []Order
    for i, p := range tc.priorities {
      o := newOrder(p, i, "", "") // We can use the index as the quantity as a cheat so our argsort works! :)
      orders = append(orders, o)
      q.Add(o)
    }

    var gotQuantities []int
    for _, o := range q {
      gotQuantities = append(gotQuantities, o.quantity)
    }

    if !reflect.DeepEqual(gotQuantities, tc.wantQuantities) {
      t.Errorf("after adding Orders %+v to empty Queue; got Queue quantities %d; want %d", orders, gotQuantities, wantQuantities)
    }
  }
}
```